tests/datasets/test_views.py

```diff
@@ -109,6 +109,21 @@ def search_datasets():
     return [dataset1, dataset2, dataset3]
 
 
+@pytest.mark.haystack
+def test_dataset_list_view_anon_user_with_datasets(app: DjangoTestApp):
+    DatasetFactory()
+    DatasetFactory()
+    DatasetFactory()
+    resp = app.get(reverse('dataset-list'))
+    assert len(resp.context['object_list']) == 3
+
+
+@pytest.mark.haystack
+def test_dataset_list_view_anon_user_without_datasets(app: DjangoTestApp):
+    resp = app.get(reverse('dataset-list'))
+    assert len(resp.context['object_list']) == 0
+
+
```

Comment:
Bent jau man šie testai praeina su ir be pataisymo. Reikėtų testų, kurie realiai atkartoja problemą. Jei iš facetų dingsta datos facetas, tai reiškia, kad kažkas iš esmės yra pagriuvę. Po atnaujinimo, ir toliau gaunu 500 klaidą.